pkg/operator/controllers/pullsecret/pullsecret_controller.go

```diff
@@ -65,85 +84,66 @@ func (r *PullSecretReconciler) Reconcile(request ctrl.Request) (ctrl.Result, err
 		return reconcile.Result{}, err
 	}
 
+	// cluster object is needed, controller changes its state based on RH pull secret presence
+	cluster, err := r.arocli.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
 	return reconcile.Result{}, retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
I really tempted to split this to separate reconcile loops. Now all this logic sits in retry loop from line 93 to 132. This make  it to retry on smallest failure. This is not good.
1. We should keep fixup loop as it is.
2.  We should add second retry loop to update RH Secret present condition (PR 1)
3. Add third loop to handle samples operator condition based on condition on 2 (PR 2).

All ran in order so we always know that step 3 is ran after step 2 and condition will be up-to-date. So we avoid spaghetti code here.

It could looks something like
```
	for _, f := range []func(context.Context) (interface{}, error){
		s.reconcilePullSecret, // must run first
		s.reconcileRegistryKeys,
		s.reconfileSamplesOperator, // must run after RegistryKeys reconciler
	} {
		o, err := f(ctx)
		if err != nil {
			s.log.Error(err)
			return reconcile.Result{}, err
		}
```